---
title: 'iSC.MEB: seqFISH Data Analysis'
author: "Xiao Zhang"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{iSC.MEB: seqFISH Data Analysis}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  tidy = TRUE,
  fig.width = 11,
  tidy.opts = list(width.cutoff = 95),
  message = FALSE,
  warning = FALSE,
  time_it = TRUE
)
all_times <- list()  # store the time for each chunk
knitr::knit_hooks$set(time_it = local({
  now <- NULL
  function(before, options) {
    if (before) {
      now <<- Sys.time()
    } else {
      res <- difftime(Sys.time(), now, units = "secs")
      all_times[[options$label]] <<- res
    }
  }
}))
```
  
# Intrduction
This vignette introduces the iSC.MEB workflow for the analysis of seqFISH datasets. The seqFISH dataset contains the high-resolution spatial map from 6 mouse embryo tissue sections generated by a modified version of the seqFISH (sequential fluorescence in situ hybridization) method which allows highly-effective cell segmentation [Lohoff T. et al., 2020](https://www.biorxiv.org/content/10.1101/2020.11.20.391896v1). The workflow consists of three steps

* Independent preprocessing and model setting
* Integrated clustering using iSC.MEB model
* Downstream analysis (i.e. visualization of clusters and embeddings, combined differential expression analysis)

We demonstrate the use of iSC.MEB to a subset of the seqFISH datasets that are [here](https://github.com/XiaoZhangryy/iSC.MEB/tree/main/vignettes_data), which can be downloaded to the current working path by the following command:
```{r eval = FALSE}
githubURL <- "https://github.com/XiaoZhangryy/iSC.MEB/blob/main/vignettes_data/seqFISH.rda?raw=true"
download.file(githubURL,"seqFISH.rda",mode='wb')
```

The package can be loaded with the command:
```{r  eval = TRUE}
library(iSC.MEB)
library(Seurat)
```

Then load datasets to R 
```{r  eval = TRUE}
load("seqFISH.rda")
```


# Fit an `iSC.MEB` model
First, we view the the spatial transcriptomics data.
```{r  eval = TRUE}
seuList ## a list including six Seurat object
```

## Prepare the `iSC.MEB` Object.
Then, we can create a iSCMEBObj object to prepare for `iSC.MEB` models. And next, we performe create adjacency matrix, dimensionality reduction and set model setting steps in turn. Here, the adjacency matrix is built based on L2 distance. 
```{r  eval = TRUE}
Genelist <- row.names(seuList[[1]])
iSCMEBObj <- CreateiSCMEBObject(seuList = seuList, customGenelist = Genelist, verbose = FALSE)
## check the number of genes/features after filtering step
iSCMEBObj@seulist
## Add adjacency matrix list for a iSCMEBObj object to prepare for iSC.MEB model fitting.
iSCMEBObj <-  CreateNeighbors(iSCMEBObj, radius.upper=10)
## run PCA to get low dimensional embeddings
iSCMEBObj <-  runPCA(iSCMEBObj, npcs = 15, pca.method = "APCA")
## Add a model setting in advance for an iSCMEBObj object. verbose = TRUE helps outputing the
## information in the algorithm.
iSCMEBObj <- SetModelParameters(iSCMEBObj, verbose = FALSE, maxIter_ICM = 4, maxIter = 20, coreNum=4)
```


## Fit iSC.MEB 
For function `iSCMEB`, users can specify the number of clusters `K` or set `K` to be an integer vector by using modified MBIC(BIC) to determine `K`. Here, we use MBIC to select optimal number of clusters, which is the default method. User can use specifying other criteria in function `SetModelParameters` or using function `SelectModel` to select model after model fitting. Here we briefly explain how to choose the parameter `c_penalty` in the modified BIC. In general, when the number of cluster is considered to be small, such as less than 10 clusters, `c_penalty` often be large, for example 5, 10. And when the number of cluster is considered to be large, `c_penalty` often ranges from 0.5 to 2. Most importantly, iSC.MEB is fast, scaling well in terms of sample size, which allow the user to tune the `c_penalty` based on their prior knowledge about the tissues or cells.
```{r  eval = TRUE}
iSCMEBObj <- iSCMEB(iSCMEBObj, K = 16:19)
iSCMEBObj <- SelectModel(iSCMEBObj, criteria="MBIC", c_penalty = 0.2)
p1 <- SelectKPlot(iSCMEBObj, criteria="MBIC")
p1
```

## Evaluate performance
The function `idents` can extract the labels provided by `iSC.MEB` method. Therefore, we can evaluate the clustering performance by some metrics, such as `ARI`. 
```{r  eval = TRUE}
LabelList <- lapply(iSCMEBObj@seulist, function(seu) seu@meta.data$celltype_mapped_refined)
ARI <- function(x,y) mclust::adjustedRandIndex(x,y)
ari_sections <- sapply(1:3, function(i) ARI(idents(iSCMEBObj)[[i]], LabelList[[i]]))
ari_all <- ARI(unlist(idents(iSCMEBObj)), unlist(LabelList))
print(ari_sections)
print(ari_all)
```

# Visualization
## Spatial scatter plot
In addition to metrics, we can use some visualization functions to measure the clustering results, such as the spatial scatter plot. 
```{r  eval = TRUE}
p2 <- SpaHeatMap(iSCMEBObj, item="cluster", plot_type="Scatter", 
    layout.dim=c(1, 3), nrow.legend=2, no_axis=TRUE, point_size=1.5)
p2
```

## t-SNE plot.
Next, user can visualize the inferred embeddings for biological effects between cell/domain types using two components from either tSNE or UMAP. Here, wo demonstrate the clustering and batch remove performance by t-SNE plot.
```{r  eval = TRUE}
iSCMEBObj <- CalculateTSNE(iSCMEBObj, reduction="iSCMEB", n_comp=2)
library(patchwork)
p3 <- LowEmbedPlot(iSCMEBObj, item="cluster", reduction="TSNE2", point_size=0.5)
p4 <- LowEmbedPlot(iSCMEBObj, item="batch", reduction="TSNE2", point_size=0.5)
p3 + p4
```

## Combined differential expression analysis
Finally, we also provide functions to fecility user do combined differential expression analysis. The `IntegrateSpaData` function can integrate multiple SRT data based on the our results, function `doDEG` provide top differential expression genes, and function `doHeatmap` plot heatmap. 
```{r  eval = TRUE}
seuInt <- IntegrateSpaData(iSCMEBObj, "mouse")
top10 <- doDEG(seuInt, topn=10)
p5 <- doHeatmap(seuInt, top10$gene)
p5
```

# Session information
```{r  eval = TRUE}
sessionInfo()
```