<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="iSC.MEB">
<title>iSC.MEB: seqFISH Data Analysis • iSC.MEB</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="iSC.MEB: seqFISH Data Analysis">
<meta property="og:description" content="iSC.MEB">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">iSC.MEB</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/iSC-MEB.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/iSC.MEB.DLPFC.html">iSC.MEB: DLPFC Data Analysis</a>
    <a class="dropdown-item" href="../articles/iSC.MEB.seqFISH.html">iSC.MEB: seqFISH Data Analysis</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="nav-link" href="https://github.com/XiaoZhangryy/iSC.MEB/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>iSC.MEB: seqFISH Data Analysis</h1>
                        <h4 data-toc-skip class="author">Xiao Zhang</h4>
            
            <h4 data-toc-skip class="date">2022-10-12</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/XiaoZhangryy/iSC.MEB/blob/HEAD/vignettes/iSC.MEB.seqFISH.Rmd"><code>vignettes/iSC.MEB.seqFISH.Rmd</code></a></small>
      <div class="d-none name"><code>iSC.MEB.seqFISH.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="intrduction">Intrduction<a class="anchor" aria-label="anchor" href="#intrduction"></a>
</h2>
<p>This vignette introduces the iSC.MEB workflow for the analysis of
seqFISH datasets. The seqFISH dataset contains the high-resolution
spatial map from 6 mouse embryo tissue sections generated by a modified
version of the seqFISH (sequential fluorescence in situ hybridization)
method which allows highly-effective cell segmentation <a href="https://www.biorxiv.org/content/10.1101/2020.11.20.391896v1" class="external-link">Lohoff
T. et al., 2020</a>. The workflow consists of three steps</p>
<ul>
<li>Independent preprocessing and model setting</li>
<li>Integrated clustering using iSC.MEB model</li>
<li>Downstream analysis (i.e. visualization of clusters and
embeddings, combined differential expression analysis)</li>
</ul>
<p>We demonstrate the use of iSC.MEB to 6 seqFISH datasets that are <a href="https://github.com/XiaoZhangryy/iSC.MEB/tree/main/vignettes_data">here</a>,
which can be downloaded to the current working path by the following
command:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">githubURL</span> <span class="op">&lt;-</span> <span class="st">"https://github.com/XiaoZhangryy/iSC.MEB/blob/main/vignettes_data/seqFISH.rda?raw=true"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">githubURL</span>, <span class="st">"seqFISH.rda"</span>, mode <span class="op">=</span> <span class="st">"wb"</span><span class="op">)</span></span></code></pre></div>
<p>The package can be loaded with the command:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/XiaoZhangryy/iSC.MEB">iSC.MEB</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span></code></pre></div>
<p>Then load datasets to R</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"seqFISH.rda"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="fit-an-isc-meb-model">Fit an <code>iSC.MEB</code> model<a class="anchor" aria-label="anchor" href="#fit-an-isc-meb-model"></a>
</h2>
<p>First, we view the the spatial transcriptomics data.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seuList</span>  <span class="co">## a list including six Seurat object</span></span>
<span><span class="co">#&gt; [[1]]</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 351 features across 10129 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (351 features, 0 variable features)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 351 features across 7642 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (351 features, 0 variable features)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[3]]</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 351 features across 7654 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (351 features, 0 variable features)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[4]]</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 351 features across 6520 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (351 features, 0 variable features)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[5]]</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 351 features across 11106 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (351 features, 0 variable features)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[6]]</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 351 features across 9373 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (351 features, 0 variable features)</span></span></code></pre></div>
<div class="section level3">
<h3 id="prepare-the-isc-meb-object-">Prepare the <code>iSC.MEB</code> Object.<a class="anchor" aria-label="anchor" href="#prepare-the-isc-meb-object-"></a>
</h3>
<p>Then, we can create a iSCMEBObj object to prepare for
<code>iSC.MEB</code> models. And next, we performe create adjacency
matrix, dimensionality reduction and set model setting steps in turn.
Here, the adjacency matrix is built based on L2 distance.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Genelist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">seuList</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">iSCMEBObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CreateiSCMEBObject.html">CreateiSCMEBObject</a></span><span class="op">(</span>seuList <span class="op">=</span> <span class="va">seuList</span>, customGenelist <span class="op">=</span> <span class="va">Genelist</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">## check the number of genes/features after filtering step</span></span>
<span><span class="va">iSCMEBObj</span><span class="op">@</span><span class="va">seulist</span></span>
<span><span class="co">#&gt; [[1]]</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 351 features across 10072 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (351 features, 0 variable features)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 351 features across 7594 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (351 features, 0 variable features)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[3]]</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 351 features across 7638 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (351 features, 0 variable features)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[4]]</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 351 features across 6499 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (351 features, 0 variable features)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[5]]</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 351 features across 10861 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (351 features, 0 variable features)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[6]]</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 351 features across 9136 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (351 features, 0 variable features)</span></span>
<span><span class="co">## Add adjacency matrix list for a iSCMEBObj object to prepare for iSC.MEB model fitting.</span></span>
<span><span class="va">iSCMEBObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CreateNeighbors.html">CreateNeighbors</a></span><span class="op">(</span><span class="va">iSCMEBObj</span>, radius.upper <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="co">## run PCA to get low dimensional embeddings</span></span>
<span><span class="va">iSCMEBObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span><span class="va">iSCMEBObj</span>, npcs <span class="op">=</span> <span class="fl">15</span>, pca.method <span class="op">=</span> <span class="st">"APCA"</span><span class="op">)</span></span>
<span><span class="co">## Add a model setting in advance for an iSCMEBObj object. verbose = TRUE helps outputing the</span></span>
<span><span class="co">## information in the algorithm.</span></span>
<span><span class="va">iSCMEBObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SetModelParameters.html">SetModelParameters</a></span><span class="op">(</span><span class="va">iSCMEBObj</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>, coreNum <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="fit-isc-meb">Fit iSC.MEB<a class="anchor" aria-label="anchor" href="#fit-isc-meb"></a>
</h3>
<p>For function <code>iSCMEB</code>, users can specify the number of
clusters <code>K</code> or set <code>K</code> to be an integer vector by
using modified MBIC(BIC) to determine <code>K</code>. Here, we use MBIC
to select optimal number of clusters, which is the default method. User
can use specifying other criteria in function
<code>SetModelParameters</code> or using function
<code>SelectModel</code> to select model after model fitting. Here we
briefly explain how to choose the parameter <code>c_penalty</code> in
the modified BIC. In general, when the number of cluster is considered
to be small, such as less than 10 clusters, <code>c_penalty</code> often
be large, for example 5, 10. And when the number of cluster is
considered to be large, <code>c_penalty</code> often ranges from 0.5 to
2. Most importantly, iSC.MEB is fast, scaling well in terms of sample
size, which allow the user to tune the <code>c_penalty</code> based on
their prior knowledge about the tissues or cells.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">iSCMEBObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/iSCMEB.html">iSCMEB</a></span><span class="op">(</span><span class="va">iSCMEBObj</span>, K <span class="op">=</span> <span class="fl">16</span><span class="op">:</span><span class="fl">23</span><span class="op">)</span></span>
<span><span class="va">iSCMEBObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SelectModel.html">SelectModel</a></span><span class="op">(</span><span class="va">iSCMEBObj</span>, criteria <span class="op">=</span> <span class="st">"MBIC"</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SelectKPlot.html">SelectKPlot</a></span><span class="op">(</span><span class="va">iSCMEBObj</span>, criteria <span class="op">=</span> <span class="st">"MBIC"</span><span class="op">)</span></span>
<span><span class="va">p1</span></span></code></pre></div>
<p><img src="iSC.MEB.seqFISH_files/figure-html/unnamed-chunk-7-1.png" width="1056"></p>
</div>
<div class="section level3">
<h3 id="evaluate-performance">Evaluate performance<a class="anchor" aria-label="anchor" href="#evaluate-performance"></a>
</h3>
<p>The function <code>idents</code> can extract the labels provided by
<code>iSC.MEB</code> method. Therefore, we can evaluate the clustering
performance by some metrics, such as <code>ARI</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">LabelList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">iSCMEBObj</span><span class="op">@</span><span class="va">seulist</span>, <span class="kw">function</span><span class="op">(</span><span class="va">seu</span><span class="op">)</span> <span class="va">seu</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">celltype_mapped_refined</span><span class="op">)</span></span>
<span><span class="va">ARI</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span><span class="op">)</span> <span class="fu">mclust</span><span class="fu">::</span><span class="fu"><a href="https://mclust-org.github.io/mclust/reference/adjustedRandIndex.html" class="external-link">adjustedRandIndex</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span><span class="op">)</span></span>
<span><span class="va">ari_sections</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, <span class="kw">function</span><span class="op">(</span><span class="va">i</span><span class="op">)</span> <span class="fu">ARI</span><span class="op">(</span><span class="fu"><a href="../reference/idents.html">idents</a></span><span class="op">(</span><span class="va">iSCMEBObj</span><span class="op">)</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, <span class="va">LabelList</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ari_all</span> <span class="op">&lt;-</span> <span class="fu">ARI</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="../reference/idents.html">idents</a></span><span class="op">(</span><span class="va">iSCMEBObj</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">LabelList</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">ari_sections</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.5204352 0.5176676 0.5523073 0.4926388 0.5420868 0.5473268</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">ari_all</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.5156581</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<div class="section level3">
<h3 id="spatial-scatter-plot">Spatial scatter plot<a class="anchor" aria-label="anchor" href="#spatial-scatter-plot"></a>
</h3>
<p>In addition to metrics, we can use some visualization functions to
measure the clustering results, such as the spatial scatter plot.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaHeatMap.html">SpaHeatMap</a></span><span class="op">(</span><span class="va">iSCMEBObj</span>, item <span class="op">=</span> <span class="st">"cluster"</span>, plot_type <span class="op">=</span> <span class="st">"Scatter"</span>, layout.dim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">2</span><span class="op">)</span>, nrow.legend <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    no_axis <span class="op">=</span> <span class="cn">TRUE</span>, point_size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">p2</span></span></code></pre></div>
<p><img src="iSC.MEB.seqFISH_files/figure-html/unnamed-chunk-9-1.png" width="1056"></p>
</div>
<div class="section level3">
<h3 id="t-sne-plot-">t-SNE plot.<a class="anchor" aria-label="anchor" href="#t-sne-plot-"></a>
</h3>
<p>Next, user can visualize the inferred embeddings for biological
effects between cell/domain types using two components from either tSNE
or UMAP. Here, wo demonstrate the clustering and batch remove
performance by t-SNE plot.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">iSCMEBObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CalculateTSNE.html">CalculateTSNE</a></span><span class="op">(</span><span class="va">iSCMEBObj</span>, reduction <span class="op">=</span> <span class="st">"iSCMEB"</span>, n_comp <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/LowEmbedPlot.html">LowEmbedPlot</a></span><span class="op">(</span><span class="va">iSCMEBObj</span>, item <span class="op">=</span> <span class="st">"cluster"</span>, reduction <span class="op">=</span> <span class="st">"TSNE2"</span>, point_size <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span>
<span><span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/LowEmbedPlot.html">LowEmbedPlot</a></span><span class="op">(</span><span class="va">iSCMEBObj</span>, item <span class="op">=</span> <span class="st">"batch"</span>, reduction <span class="op">=</span> <span class="st">"TSNE2"</span>, point_size <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span>
<span><span class="va">p3</span> <span class="op">+</span> <span class="va">p4</span></span></code></pre></div>
<p><img src="iSC.MEB.seqFISH_files/figure-html/unnamed-chunk-10-1.png" width="1056"></p>
</div>
<div class="section level3">
<h3 id="combined-differential-expression-analysis">Combined differential expression analysis<a class="anchor" aria-label="anchor" href="#combined-differential-expression-analysis"></a>
</h3>
<p>Finally, we also provide functions to fecility user do combined
differential expression analysis. The <code>IntegrateSpaData</code>
function can integrate multiple SRT data based on the our results,
function <code>doDEG</code> provide top differential expression genes,
and function <code>doHeatmap</code> plot heatmap.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seuInt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/IntegrateSpaData.html">IntegrateSpaData</a></span><span class="op">(</span><span class="va">iSCMEBObj</span>, <span class="st">"mouse"</span><span class="op">)</span></span>
<span><span class="va">top10</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doDEG.html">doDEG</a></span><span class="op">(</span><span class="va">seuInt</span>, topn <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">p5</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doHeatmap.html">doHeatmap</a></span><span class="op">(</span><span class="va">seuInt</span>, <span class="va">top10</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span></span>
<span><span class="va">p5</span></span></code></pre></div>
<p><img src="iSC.MEB.seqFISH_files/figure-html/unnamed-chunk-11-1.png" width="1056"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.1.3 (2022-03-10)</span></span>
<span><span class="co">#&gt; Platform: x86_64-conda-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: CentOS Linux 7 (Core)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS/LAPACK: /home/Zhangxiao/anaconda3/envs/r41/lib/libopenblasp-r0.3.20.so</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] patchwork_1.1.2    sp_1.5-0           SeuratObject_4.1.1 Seurat_4.1.1      </span></span>
<span><span class="co">#&gt; [5] iSC.MEB_1.0        ggplot2_3.3.6      gtools_3.9.3      </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] utf8_1.2.2                  reticulate_1.25            </span></span>
<span><span class="co">#&gt;   [3] tidyselect_1.1.2            htmlwidgets_1.5.4          </span></span>
<span><span class="co">#&gt;   [5] grid_4.1.3                  BiocParallel_1.28.3        </span></span>
<span><span class="co">#&gt;   [7] Rtsne_0.16                  ScaledMatrix_1.2.0         </span></span>
<span><span class="co">#&gt;   [9] munsell_0.5.0               codetools_0.2-18           </span></span>
<span><span class="co">#&gt;  [11] ragg_1.2.2                  ica_1.0-3                  </span></span>
<span><span class="co">#&gt;  [13] future_1.27.0               miniUI_0.1.1.1             </span></span>
<span><span class="co">#&gt;  [15] withr_2.5.0                 spatstat.random_2.2-0      </span></span>
<span><span class="co">#&gt;  [17] colorspace_2.0-3            progressr_0.10.1           </span></span>
<span><span class="co">#&gt;  [19] Biobase_2.54.0              highr_0.9                  </span></span>
<span><span class="co">#&gt;  [21] knitr_1.40                  stats4_4.1.3               </span></span>
<span><span class="co">#&gt;  [23] SingleCellExperiment_1.16.0 ROCR_1.0-11                </span></span>
<span><span class="co">#&gt;  [25] ggsignif_0.6.3              tensor_1.5                 </span></span>
<span><span class="co">#&gt;  [27] listenv_0.8.0               labeling_0.4.2             </span></span>
<span><span class="co">#&gt;  [29] MatrixGenerics_1.6.0        GenomeInfoDbData_1.2.7     </span></span>
<span><span class="co">#&gt;  [31] polyclip_1.10-0             farver_2.1.1               </span></span>
<span><span class="co">#&gt;  [33] rprojroot_2.0.3             parallelly_1.32.1          </span></span>
<span><span class="co">#&gt;  [35] vctrs_0.4.1                 generics_0.1.3             </span></span>
<span><span class="co">#&gt;  [37] xfun_0.32                   R6_2.5.1                   </span></span>
<span><span class="co">#&gt;  [39] GenomeInfoDb_1.30.1         ggbeeswarm_0.6.0           </span></span>
<span><span class="co">#&gt;  [41] rsvd_1.0.5                  bitops_1.0-7               </span></span>
<span><span class="co">#&gt;  [43] spatstat.utils_2.3-1        cachem_1.0.6               </span></span>
<span><span class="co">#&gt;  [45] DelayedArray_0.20.0         assertthat_0.2.1           </span></span>
<span><span class="co">#&gt;  [47] promises_1.2.0.1            scales_1.2.1               </span></span>
<span><span class="co">#&gt;  [49] beeswarm_0.4.0              rgeos_0.5-9                </span></span>
<span><span class="co">#&gt;  [51] gtable_0.3.1                beachmat_2.10.0            </span></span>
<span><span class="co">#&gt;  [53] globals_0.16.1              goftest_1.2-3              </span></span>
<span><span class="co">#&gt;  [55] rlang_1.0.6                 systemfonts_1.0.4          </span></span>
<span><span class="co">#&gt;  [57] splines_4.1.3               rstatix_0.7.0              </span></span>
<span><span class="co">#&gt;  [59] lazyeval_0.2.2              spatstat.geom_2.4-0        </span></span>
<span><span class="co">#&gt;  [61] broom_1.0.1                 yaml_2.3.5                 </span></span>
<span><span class="co">#&gt;  [63] reshape2_1.4.4              abind_1.4-5                </span></span>
<span><span class="co">#&gt;  [65] backports_1.4.1             httpuv_1.6.5               </span></span>
<span><span class="co">#&gt;  [67] tools_4.1.3                 ellipsis_0.3.2             </span></span>
<span><span class="co">#&gt;  [69] spatstat.core_2.4-4         jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt;  [71] RColorBrewer_1.1-3          BiocGenerics_0.40.0        </span></span>
<span><span class="co">#&gt;  [73] ggridges_0.5.3              Rcpp_1.0.9                 </span></span>
<span><span class="co">#&gt;  [75] plyr_1.8.7                  sparseMatrixStats_1.6.0    </span></span>
<span><span class="co">#&gt;  [77] zlibbioc_1.40.0             purrr_0.3.4                </span></span>
<span><span class="co">#&gt;  [79] RCurl_1.98-1.8              ggpubr_0.4.0               </span></span>
<span><span class="co">#&gt;  [81] rpart_4.1.16                deldir_1.0-6               </span></span>
<span><span class="co">#&gt;  [83] viridis_0.6.2               pbapply_1.5-0              </span></span>
<span><span class="co">#&gt;  [85] cowplot_1.1.1               S4Vectors_0.32.4           </span></span>
<span><span class="co">#&gt;  [87] zoo_1.8-10                  SummarizedExperiment_1.24.0</span></span>
<span><span class="co">#&gt;  [89] ggrepel_0.9.1               cluster_2.1.4              </span></span>
<span><span class="co">#&gt;  [91] fs_1.5.2                    magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;  [93] GiRaF_1.0.1                 data.table_1.14.2          </span></span>
<span><span class="co">#&gt;  [95] scattermore_0.8             lmtest_0.9-40              </span></span>
<span><span class="co">#&gt;  [97] RANN_2.6.1                  fitdistrplus_1.1-8         </span></span>
<span><span class="co">#&gt;  [99] matrixStats_0.62.0          mime_0.12                  </span></span>
<span><span class="co">#&gt; [101] evaluate_0.16               xtable_1.8-4               </span></span>
<span><span class="co">#&gt; [103] mclust_5.4.10               IRanges_2.28.0             </span></span>
<span><span class="co">#&gt; [105] gridExtra_2.3               compiler_4.1.3             </span></span>
<span><span class="co">#&gt; [107] scater_1.22.0               tibble_3.1.8               </span></span>
<span><span class="co">#&gt; [109] KernSmooth_2.23-20          htmltools_0.5.3            </span></span>
<span><span class="co">#&gt; [111] mgcv_1.8-40                 later_1.3.0                </span></span>
<span><span class="co">#&gt; [113] tidyr_1.2.1                 DBI_1.1.3                  </span></span>
<span><span class="co">#&gt; [115] formatR_1.12                MASS_7.3-58.1              </span></span>
<span><span class="co">#&gt; [117] Matrix_1.4-1                car_3.1-0                  </span></span>
<span><span class="co">#&gt; [119] cli_3.4.0                   parallel_4.1.3             </span></span>
<span><span class="co">#&gt; [121] igraph_1.3.5                DR.SC_2.9                  </span></span>
<span><span class="co">#&gt; [123] GenomicRanges_1.46.1        pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt; [125] pkgdown_2.0.6               plotly_4.10.0              </span></span>
<span><span class="co">#&gt; [127] scuttle_1.4.0               spatstat.sparse_2.1-1      </span></span>
<span><span class="co">#&gt; [129] vipor_0.4.5                 bslib_0.4.0                </span></span>
<span><span class="co">#&gt; [131] XVector_0.34.0              CompQuadForm_1.4.3         </span></span>
<span><span class="co">#&gt; [133] stringr_1.4.1               digest_0.6.29              </span></span>
<span><span class="co">#&gt; [135] sctransform_0.3.4           RcppAnnoy_0.0.19           </span></span>
<span><span class="co">#&gt; [137] spatstat.data_2.2-0         rmarkdown_2.16             </span></span>
<span><span class="co">#&gt; [139] leiden_0.4.2                uwot_0.1.14                </span></span>
<span><span class="co">#&gt; [141] DelayedMatrixStats_1.16.0   shiny_1.7.2                </span></span>
<span><span class="co">#&gt; [143] lifecycle_1.0.1             nlme_3.1-159               </span></span>
<span><span class="co">#&gt; [145] jsonlite_1.8.0              carData_3.0-5              </span></span>
<span><span class="co">#&gt; [147] BiocNeighbors_1.12.0        limma_3.50.3               </span></span>
<span><span class="co">#&gt; [149] desc_1.4.1                  viridisLite_0.4.1          </span></span>
<span><span class="co">#&gt; [151] fansi_1.0.3                 pillar_1.8.1               </span></span>
<span><span class="co">#&gt; [153] lattice_0.20-45             fastmap_1.1.0              </span></span>
<span><span class="co">#&gt; [155] httr_1.4.4                  survival_3.4-0             </span></span>
<span><span class="co">#&gt; [157] glue_1.6.2                  png_0.1-7                  </span></span>
<span><span class="co">#&gt; [159] stringi_1.7.8               sass_0.4.2                 </span></span>
<span><span class="co">#&gt; [161] textshaping_0.3.6           BiocSingular_1.10.0        </span></span>
<span><span class="co">#&gt; [163] memoise_2.0.1               dplyr_1.0.10               </span></span>
<span><span class="co">#&gt; [165] irlba_2.3.5                 future.apply_1.9.0</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Xiao Zhang, Wei Liu, Jin Liu.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
