<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="iSC.MEB">
<title>iSC.MEB: DLPFC Data Analysis • iSC.MEB</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="iSC.MEB: DLPFC Data Analysis">
<meta property="og:description" content="iSC.MEB">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">iSC.MEB</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/iSC-MEB.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/iSC.MEB.DLPFC.html">iSC.MEB: DLPFC Data Analysis</a>
    <a class="dropdown-item" href="../articles/iSC.MEB.seqFISH.html">iSC.MEB: seqFISH Data Analysis</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="nav-link" href="https://github.com/XiaoZhangryy/iSC.MEB/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>iSC.MEB: DLPFC Data Analysis</h1>
                        <h4 data-toc-skip class="author">Xiao Zhang</h4>
            
            <h4 data-toc-skip class="date">2022-12-15</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/XiaoZhangryy/iSC.MEB/blob/HEAD/vignettes/iSC.MEB.DLPFC.Rmd"><code>vignettes/iSC.MEB.DLPFC.Rmd</code></a></small>
      <div class="d-none name"><code>iSC.MEB.DLPFC.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="intrduction">Intrduction<a class="anchor" aria-label="anchor" href="#intrduction"></a>
</h2>
<p>This vignette introduces the <code>iSC.MEB</code> workflow for the
analysis of the DLPFC dataset <a href="https://www.nature.com/articles/s41593-020-00787-0" class="external-link">Maynard K. et
al, 2021</a>, which contains the spatial topography of gene expression
from 12 human postmortem DLPFC tissue sections generated by using the
10x Genomics Visium platform. The workflow consists of three steps</p>
<ul>
<li>Independent preprocessing and model setting</li>
<li>Integrated clustering using <code>iSC.MEB</code> model</li>
<li>Downstream analysis (i.e. visualization of clusters and
embeddings)</li>
</ul>
<p>We demonstrate the use of <code>iSC.MEB</code> to a subset of the
DLPFC data after filter that are <a href="https://github.com/XiaoZhangryy/iSC.MEB/tree/main/vignettes_data">here</a>,
which can be downloaded to the current working path by the following
command:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">githubURL</span> <span class="op">&lt;-</span> <span class="st">"https://github.com/XiaoZhangryy/iSC.MEB/blob/main/vignettes_data/DLPFC.rda?raw=true"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">githubURL</span>, <span class="st">"DLPFC.rda"</span>, mode <span class="op">=</span> <span class="st">"wb"</span><span class="op">)</span></span></code></pre></div>
<p>The package can be loaded with the command:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/XiaoZhangryy/iSC.MEB">iSC.MEB</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span></code></pre></div>
<p>Then load datasets to R</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"DLPFC.rda"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="fit-an-isc-meb-model">Fit an <code>iSC.MEB</code> model<a class="anchor" aria-label="anchor" href="#fit-an-isc-meb-model"></a>
</h2>
<p>First, we view the the spatial transcriptomics data with Visium
platform.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seuList</span>  <span class="co">## a list including two Seurat object</span></span>
<span><span class="co">#&gt; $`151507`</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 2000 features across 4226 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (2000 features, 0 variable features)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`151508`</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 2000 features across 4384 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (2000 features, 0 variable features)</span></span></code></pre></div>
<div class="section level3">
<h3 id="prepare-the-isc-meb-object-">Prepare the <code>iSC.MEB</code> Object.<a class="anchor" aria-label="anchor" href="#prepare-the-isc-meb-object-"></a>
</h3>
<p>Then, we can create a iSCMEBObj object to prepare for
<code>iSC.MEB</code> models. And next, we performe create adjacency
matrix, dimensionality reduction and set model setting steps in
turn.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">iSCMEBObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CreateiSCMEBObject.html">CreateiSCMEBObject</a></span><span class="op">(</span>seuList <span class="op">=</span> <span class="va">seuList</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>, premin.spots <span class="op">=</span> <span class="fl">0</span>, postmin.spots <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="co">## check the number of genes/features after filtering step</span></span>
<span><span class="va">iSCMEBObj</span><span class="op">@</span><span class="va">seulist</span></span>
<span><span class="co">#&gt; $`151507`</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 2000 features across 4226 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (2000 features, 2000 variable features)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`151508`</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 2000 features across 4384 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (2000 features, 2000 variable features)</span></span>
<span><span class="co">## Add adjacency matrix list for a iSCMEBObj object to prepare for iSC.MEB model fitting.</span></span>
<span><span class="va">iSCMEBObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CreateNeighbors.html">CreateNeighbors</a></span><span class="op">(</span><span class="va">iSCMEBObj</span>, platform <span class="op">=</span> <span class="st">"Visium"</span><span class="op">)</span></span>
<span><span class="co">## run PCA to get low dimensional embeddings</span></span>
<span><span class="va">iSCMEBObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span><span class="va">iSCMEBObj</span>, npcs <span class="op">=</span> <span class="fl">15</span>, pca.method <span class="op">=</span> <span class="st">"APCA"</span><span class="op">)</span></span>
<span><span class="co">## Add a model setting in advance for an iSCMEBObj object. verbose = TRUE helps outputing the</span></span>
<span><span class="co">## information in the algorithm.</span></span>
<span><span class="va">iSCMEBObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SetModelParameters.html">SetModelParameters</a></span><span class="op">(</span><span class="va">iSCMEBObj</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="fit-isc-meb">Fit iSC.MEB<a class="anchor" aria-label="anchor" href="#fit-isc-meb"></a>
</h3>
<p>For function <code>iSCMEB</code>, users can specify the number of
clusters <code>K</code> or set <code>K</code> to be an integer vector by
using modified MBIC(BIC) to determine <code>K</code>. Here, we use
user-specified number of clusters.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Given K</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>iSCMEBObj <span class="ot">&lt;-</span> <span class="fu">iSCMEB</span>(iSCMEBObj, <span class="at">K =</span> <span class="dv">7</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; fitting ...</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="er">|</span>                                                                      <span class="er">|</span>   <span class="dv">0</span>%</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===================================</span>                                   <span class="er">|</span>  <span class="dv">50</span>%</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======================================================================|</span> <span class="dv">100</span>%</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Finish variable initialization </span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 2, loglik= -143673.140702, dloglik=0.999933 </span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 3, loglik= -135584.966524, dloglik=0.056296 </span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 4, loglik= -130922.056202, dloglik=0.034391 </span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 5, loglik= -127348.034983, dloglik=0.027299 </span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 6, loglik= -124686.996633, dloglik=0.020896 </span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 7, loglik= -122764.719220, dloglik=0.015417 </span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 8, loglik= -121388.856896, dloglik=0.011207 </span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 9, loglik= -120372.623719, dloglik=0.008372 </span></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 10, loglik= -119594.151740, dloglik=0.006467 </span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 11, loglik= -118978.326627, dloglik=0.005149 </span></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 12, loglik= -118476.812547, dloglik=0.004215 </span></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 13, loglik= -118053.111872, dloglik=0.003576 </span></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 14, loglik= -117697.998012, dloglik=0.003008 </span></span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 15, loglik= -117402.318720, dloglik=0.002512 </span></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 16, loglik= -117156.949410, dloglik=0.002090 </span></span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 17, loglik= -116959.900276, dloglik=0.001682 </span></span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 18, loglik= -116787.632145, dloglik=0.001473 </span></span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 19, loglik= -116657.224113, dloglik=0.001117 </span></span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 20, loglik= -116544.747450, dloglik=0.000964 </span></span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 21, loglik= -116450.350990, dloglik=0.000810 </span></span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 22, loglik= -116369.634045, dloglik=0.000693 </span></span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 23, loglik= -116289.419857, dloglik=0.000689 </span></span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 24, loglik= -116184.897306, dloglik=0.000899 </span></span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; K = 7, iter = 25, loglik= -116131.167896, dloglik=0.000462</span></span></code></pre></div>
<p>The function <code>iSCMEB</code> will automatically choose the
optimal path by the <code>criteria</code> specified in setting model
parameters step. User can slao using function <code>SelectModel</code>
to select model.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">iSCMEBObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SelectModel.html">SelectModel</a></span><span class="op">(</span><span class="va">iSCMEBObj</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="evaluate-performance">Evaluate performance<a class="anchor" aria-label="anchor" href="#evaluate-performance"></a>
</h3>
<p>The function <code>idents</code> can extract the labels provided by
<code>iSC.MEB</code> method. Therefore, we can evaluate the clustering
performance by some metrics, such as <code>ARI</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">LabelList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">iSCMEBObj</span><span class="op">@</span><span class="va">seulist</span>, <span class="kw">function</span><span class="op">(</span><span class="va">seu</span><span class="op">)</span> <span class="va">seu</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">layer_guess_reordered</span><span class="op">)</span></span>
<span><span class="va">ARI</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span><span class="op">)</span> <span class="fu">mclust</span><span class="fu">::</span><span class="fu"><a href="https://mclust-org.github.io/mclust/reference/adjustedRandIndex.html" class="external-link">adjustedRandIndex</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span><span class="op">)</span></span>
<span><span class="va">ari_sections</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="kw">function</span><span class="op">(</span><span class="va">i</span><span class="op">)</span> <span class="fu">ARI</span><span class="op">(</span><span class="fu"><a href="../reference/idents.html">idents</a></span><span class="op">(</span><span class="va">iSCMEBObj</span><span class="op">)</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, <span class="va">LabelList</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ari_all</span> <span class="op">&lt;-</span> <span class="fu">ARI</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="../reference/idents.html">idents</a></span><span class="op">(</span><span class="va">iSCMEBObj</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">LabelList</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">ari_sections</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.5425156 0.5096009</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">ari_all</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.520498</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<div class="section level3">
<h3 id="spatial-scatter-plot">Spatial scatter plot<a class="anchor" aria-label="anchor" href="#spatial-scatter-plot"></a>
</h3>
<p>In addition to metrics, we can use some visualization functions to
measure the clustering results, such as the spatial scatter plot.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cols</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#fb8072"</span>, <span class="st">"#bebada"</span>, <span class="st">"#80b1d3"</span>, <span class="st">"#ffffb3"</span>, <span class="st">"#8dd3c7"</span>, <span class="st">"#b3de69"</span>, <span class="st">"#fdb462"</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaHeatMap.html">SpaHeatMap</a></span><span class="op">(</span><span class="va">iSCMEBObj</span>, item <span class="op">=</span> <span class="st">"cluster"</span>, plot_type <span class="op">=</span> <span class="st">"Scatter"</span>, nrow.legend <span class="op">=</span> <span class="fl">1</span>, layout.dim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,</span>
<span>    <span class="fl">2</span><span class="op">)</span>, no_axis <span class="op">=</span> <span class="cn">TRUE</span>, cols <span class="op">=</span> <span class="va">cols</span><span class="op">)</span></span>
<span><span class="va">p1</span></span></code></pre></div>
<p><img src="iSC.MEB.DLPFC_files/figure-html/unnamed-chunk-10-1.png" width="1056"></p>
</div>
<div class="section level3">
<h3 id="t-sne-plot-">t-SNE plot.<a class="anchor" aria-label="anchor" href="#t-sne-plot-"></a>
</h3>
<p>Next, user can visualize the inferred embeddings for biological
effects between cell/domain types using two components from either tSNE
or UMAP. Here, wo demonstrate the clustering and batch remove
performance by t-SNE plot.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">iSCMEBObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CalculateTSNE.html">CalculateTSNE</a></span><span class="op">(</span><span class="va">iSCMEBObj</span>, reduction <span class="op">=</span> <span class="st">"iSCMEB"</span>, n_comp <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/LowEmbedPlot.html">LowEmbedPlot</a></span><span class="op">(</span><span class="va">iSCMEBObj</span>, item <span class="op">=</span> <span class="st">"cluster"</span>, reduction <span class="op">=</span> <span class="st">"TSNE2"</span>, point_size <span class="op">=</span> <span class="fl">0.3</span><span class="op">)</span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/LowEmbedPlot.html">LowEmbedPlot</a></span><span class="op">(</span><span class="va">iSCMEBObj</span>, item <span class="op">=</span> <span class="st">"batch"</span>, reduction <span class="op">=</span> <span class="st">"TSNE2"</span>, point_size <span class="op">=</span> <span class="fl">0.3</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">+</span> <span class="va">p3</span></span></code></pre></div>
<p><img src="iSC.MEB.DLPFC_files/figure-html/unnamed-chunk-11-1.png" width="1056"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.1.3 (2022-03-10)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Windows 10 x64 (build 19045)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=English_United States.1252 </span></span>
<span><span class="co">#&gt; [2] LC_CTYPE=English_United States.1252   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=English_United States.1252</span></span>
<span><span class="co">#&gt; [4] LC_NUMERIC=C                          </span></span>
<span><span class="co">#&gt; [5] LC_TIME=English_United States.1252    </span></span>
<span><span class="co">#&gt; system code page: 936</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] patchwork_1.1.2    SeuratObject_4.1.3 Seurat_4.2.1       iSC.MEB_1.0       </span></span>
<span><span class="co">#&gt; [5] ggplot2_3.4.0      gtools_3.9.3      </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] utf8_1.2.2                  spatstat.explore_3.0-5     </span></span>
<span><span class="co">#&gt;   [3] reticulate_1.26             tidyselect_1.2.0           </span></span>
<span><span class="co">#&gt;   [5] htmlwidgets_1.5.4           grid_4.1.3                 </span></span>
<span><span class="co">#&gt;   [7] BiocParallel_1.28.3         Rtsne_0.16                 </span></span>
<span><span class="co">#&gt;   [9] ScaledMatrix_1.2.0          munsell_0.5.0              </span></span>
<span><span class="co">#&gt;  [11] codetools_0.2-18            ragg_1.2.4                 </span></span>
<span><span class="co">#&gt;  [13] ica_1.0-3                   future_1.29.0              </span></span>
<span><span class="co">#&gt;  [15] miniUI_0.1.1.1              withr_2.5.0                </span></span>
<span><span class="co">#&gt;  [17] spatstat.random_3.0-1       colorspace_2.0-3           </span></span>
<span><span class="co">#&gt;  [19] progressr_0.11.0            Biobase_2.54.0             </span></span>
<span><span class="co">#&gt;  [21] highr_0.9                   knitr_1.41                 </span></span>
<span><span class="co">#&gt;  [23] rstudioapi_0.14             stats4_4.1.3               </span></span>
<span><span class="co">#&gt;  [25] SingleCellExperiment_1.16.0 ROCR_1.0-11                </span></span>
<span><span class="co">#&gt;  [27] ggsignif_0.6.4              tensor_1.5                 </span></span>
<span><span class="co">#&gt;  [29] listenv_0.8.0               labeling_0.4.2             </span></span>
<span><span class="co">#&gt;  [31] MatrixGenerics_1.6.0        GenomeInfoDbData_1.2.7     </span></span>
<span><span class="co">#&gt;  [33] polyclip_1.10-4             farver_2.1.1               </span></span>
<span><span class="co">#&gt;  [35] rprojroot_2.0.3             parallelly_1.32.1          </span></span>
<span><span class="co">#&gt;  [37] vctrs_0.5.0                 generics_0.1.3             </span></span>
<span><span class="co">#&gt;  [39] xfun_0.35                   R6_2.5.1                   </span></span>
<span><span class="co">#&gt;  [41] GenomeInfoDb_1.30.1         ggbeeswarm_0.6.0           </span></span>
<span><span class="co">#&gt;  [43] rsvd_1.0.5                  bitops_1.0-7               </span></span>
<span><span class="co">#&gt;  [45] spatstat.utils_3.0-1        cachem_1.0.6               </span></span>
<span><span class="co">#&gt;  [47] DelayedArray_0.20.0         assertthat_0.2.1           </span></span>
<span><span class="co">#&gt;  [49] promises_1.2.0.1            scales_1.2.1               </span></span>
<span><span class="co">#&gt;  [51] beeswarm_0.4.0              gtable_0.3.1               </span></span>
<span><span class="co">#&gt;  [53] beachmat_2.10.0             globals_0.16.1             </span></span>
<span><span class="co">#&gt;  [55] goftest_1.2-3               rlang_1.0.6                </span></span>
<span><span class="co">#&gt;  [57] systemfonts_1.0.4           splines_4.1.3              </span></span>
<span><span class="co">#&gt;  [59] rstatix_0.7.1               lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt;  [61] spatstat.geom_3.0-3         broom_1.0.1                </span></span>
<span><span class="co">#&gt;  [63] yaml_2.3.6                  reshape2_1.4.4             </span></span>
<span><span class="co">#&gt;  [65] abind_1.4-5                 backports_1.4.1            </span></span>
<span><span class="co">#&gt;  [67] httpuv_1.6.6                tools_4.1.3                </span></span>
<span><span class="co">#&gt;  [69] ellipsis_0.3.2              jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt;  [71] RColorBrewer_1.1-3          BiocGenerics_0.40.0        </span></span>
<span><span class="co">#&gt;  [73] ggridges_0.5.4              Rcpp_1.0.9                 </span></span>
<span><span class="co">#&gt;  [75] plyr_1.8.8                  sparseMatrixStats_1.6.0    </span></span>
<span><span class="co">#&gt;  [77] zlibbioc_1.40.0             purrr_0.3.5                </span></span>
<span><span class="co">#&gt;  [79] RCurl_1.98-1.9              ggpubr_0.4.0               </span></span>
<span><span class="co">#&gt;  [81] deldir_1.0-6                viridis_0.6.2              </span></span>
<span><span class="co">#&gt;  [83] pbapply_1.5-0               cowplot_1.1.1              </span></span>
<span><span class="co">#&gt;  [85] S4Vectors_0.32.4            zoo_1.8-11                 </span></span>
<span><span class="co">#&gt;  [87] SummarizedExperiment_1.24.0 ggrepel_0.9.2              </span></span>
<span><span class="co">#&gt;  [89] cluster_2.1.4               fs_1.5.2                   </span></span>
<span><span class="co">#&gt;  [91] magrittr_2.0.3              GiRaF_1.0.1                </span></span>
<span><span class="co">#&gt;  [93] data.table_1.14.4           scattermore_0.8            </span></span>
<span><span class="co">#&gt;  [95] lmtest_0.9-40               RANN_2.6.1                 </span></span>
<span><span class="co">#&gt;  [97] fitdistrplus_1.1-8          matrixStats_0.62.0         </span></span>
<span><span class="co">#&gt;  [99] mime_0.12                   evaluate_0.18              </span></span>
<span><span class="co">#&gt; [101] xtable_1.8-4                mclust_6.0.0               </span></span>
<span><span class="co">#&gt; [103] IRanges_2.28.0              gridExtra_2.3              </span></span>
<span><span class="co">#&gt; [105] compiler_4.1.3              scater_1.22.0              </span></span>
<span><span class="co">#&gt; [107] tibble_3.1.8                KernSmooth_2.23-20         </span></span>
<span><span class="co">#&gt; [109] htmltools_0.5.3             later_1.3.0                </span></span>
<span><span class="co">#&gt; [111] tidyr_1.2.1                 DBI_1.1.3                  </span></span>
<span><span class="co">#&gt; [113] formatR_1.12                MASS_7.3-58.1              </span></span>
<span><span class="co">#&gt; [115] Matrix_1.5-1                car_3.1-1                  </span></span>
<span><span class="co">#&gt; [117] cli_3.4.1                   parallel_4.1.3             </span></span>
<span><span class="co">#&gt; [119] igraph_1.3.5                DR.SC_2.9                  </span></span>
<span><span class="co">#&gt; [121] GenomicRanges_1.46.1        pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt; [123] pkgdown_2.0.6               sp_1.5-1                   </span></span>
<span><span class="co">#&gt; [125] plotly_4.10.1               scuttle_1.4.0              </span></span>
<span><span class="co">#&gt; [127] spatstat.sparse_3.0-0       vipor_0.4.5                </span></span>
<span><span class="co">#&gt; [129] bslib_0.4.1                 XVector_0.34.0             </span></span>
<span><span class="co">#&gt; [131] CompQuadForm_1.4.3          stringr_1.4.1              </span></span>
<span><span class="co">#&gt; [133] digest_0.6.30               sctransform_0.3.5          </span></span>
<span><span class="co">#&gt; [135] RcppAnnoy_0.0.20            spatstat.data_3.0-0        </span></span>
<span><span class="co">#&gt; [137] rmarkdown_2.18              leiden_0.4.3               </span></span>
<span><span class="co">#&gt; [139] uwot_0.1.14                 DelayedMatrixStats_1.16.0  </span></span>
<span><span class="co">#&gt; [141] shiny_1.7.3                 lifecycle_1.0.3            </span></span>
<span><span class="co">#&gt; [143] nlme_3.1-160                jsonlite_1.8.3             </span></span>
<span><span class="co">#&gt; [145] carData_3.0-5               BiocNeighbors_1.12.0       </span></span>
<span><span class="co">#&gt; [147] desc_1.4.2                  viridisLite_0.4.1          </span></span>
<span><span class="co">#&gt; [149] fansi_1.0.3                 pillar_1.8.1               </span></span>
<span><span class="co">#&gt; [151] lattice_0.20-45             fastmap_1.1.0              </span></span>
<span><span class="co">#&gt; [153] httr_1.4.4                  survival_3.4-0             </span></span>
<span><span class="co">#&gt; [155] glue_1.6.2                  png_0.1-7                  </span></span>
<span><span class="co">#&gt; [157] stringi_1.7.8               sass_0.4.2                 </span></span>
<span><span class="co">#&gt; [159] textshaping_0.3.6           BiocSingular_1.10.0        </span></span>
<span><span class="co">#&gt; [161] memoise_2.0.1               dplyr_1.0.10               </span></span>
<span><span class="co">#&gt; [163] irlba_2.3.5.1               future.apply_1.10.0</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Xiao Zhang, Wei Liu, Jin Liu.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
